### PR TITLE
Add SmartRTB adapter

### PR DESCRIPTION
**Pull request by evanmsmrtb**
Opening pull request to add a prebid server entry for SmartRTB. Includes docs, tested user sync, and ~93% coverage.

### CODE REVIEW

File: adapters/smartrtb/smartrtbtest/supplemental/invalid-bid-format.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                  }
+                }
+              ]
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "expectedMakeBidsErrors": ["Unsupported creative type ALIEN_FORMAT."]
```
Comment (SyntaxNode) [inline adapters/smartrtb/smartrtbtest/supplemental/invalid-bid-format.json]: Please rebase and update to the testSpecExpectedError format, ie:
```
"expectedMakeBidsErrors": [
  {
      "value": "Unsupported creative type ALIEN_FORMAT.",
      "comparison": "literal"
  }
]
```

File: adapters/smartrtb/smartrtbtest/supplemental/non-http-ok.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          }
+        }
+      },
+      "mockResponse": {
+        "status": 500
+      }
+    }
+  ],
+
+  "expectedMakeBidsErrors": ["Unexpected HTTP status 500."]
```
Comment (SyntaxNode) [inline adapters/smartrtb/smartrtbtest/supplemental/non-http-ok.json]: Please rebase and update to the testSpecExpectedError format, ie:
```
"expectedMakeBidsErrors": [
  {
      "value": "Unexpected HTTP status 500.",
      "comparison": "literal"
  }
]
```

File: adapters/smartrtb/smartrtbtest/supplemental/bad-bidder-ext.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          ],
+          "w": 300,
+          "h": 250
+        },
+        "ext": null
+      }
+    ]
+  },
+
+  "expectedMakeRequestsErrors": ["unexpected end of JSON input"]
```
Comment (SyntaxNode) [inline adapters/smartrtb/smartrtbtest/supplemental/bad-bidder-ext.json]: Please rebase and update to the testSpecExpectedError format, ie:
```
"expectedMakeRequestsErrors": [
  {
      "value": "unexpected end of JSON input",
      "comparison": "literal"
  }
]
```

File: adapters/smartrtb/smartrtbtest/supplemental/bad-pub-value.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            "pub_id": 0,
+            "zone_id": "N4zTDq3PPEHBIODv7cXK",
+            "force_bid": true
+          }
+        }
+      }
+    ]
+  },
+
+  "expectedMakeRequestsErrors": ["json: cannot unmarshal number into Go struct field ExtImpSmartRTB.pub_id of type string"]
```
Comment (SyntaxNode) [inline adapters/smartrtb/smartrtbtest/supplemental/bad-pub-value.json]: Please rebase and update to the testSpecExpectedError format, ie:
```
"expectedMakeRequestsErrors": [
  {
      "value": "json: cannot unmarshal number into Go struct field ExtImpSmartRTB.pub_id of type string",
      "comparison": "literal"
  }
]
```

**Comment by evanmsmrtb** [comment]
This has been completed. I am unable to test as the EMX adapter always fails the tests, last time I had to manually edit the test script to skip their adapter to run our own tests. But the requested changes have been made.

**Comment by evanmsmrtb** [comment]
Good catch - this has now been corrected

**Comment by evanmsmrtb** [comment]
Checking on status.

**Comment by guscarreon** [comment]
@evanmsmrtb the PR ius not passing the Travis tests. When executing `./validate.sh` in the `USER/prebid/prebid-server` directory, the following errors seem to appear:
```
ok  	github.com/prebid/prebid-server/adapters	0.008s	coverage: 81.2% of statements

--- FAIL: TestValidParams (0.00s)

    params_test.go:18: Failed to fetch the json-schemas. Failed to load json schema at /home/travis/gopath/src/github.com/prebid/prebid-server/static/bidder-params/smartrtb.json: has a primitive type that is NOT VALID -- given: /bool/ Expected valid values are:[array boolean integer number null object string]

--- FAIL: TestInvalidParams (0.00s)

    params_test.go:32: Failed to fetch the json-schemas. Failed to load json schema at /home/travis/gopath/src/github.com/prebid/prebid-server/static/bidder-params/smartrtb.json: has a primitive type that is NOT VALID -- given: /bool/ Expected valid values are:[array boolean integer number null object string]

FAIL

coverage: 83.0% of statements

FAIL	github.com/prebid/prebid-server/adapters/33across	0.015s

The command "./validate.sh --nofmt --cov --race 10" exited with 1.
```

**Comment by guscarreon** [comment]
Hi @evanmsmrtb your adapter looks good to me, could you please correct the conflicting files issue and the test case fail that Travis found? Your PR seems ready for production after sorting these two minuscule issues.

**Comment by SyntaxNode** [comment]
@evanmsmrtb Do you think you'll be able to fix the merge conflicts? The reviews look good and I'd like to merge this in.

**Comment by evanmsmrtb** [comment]
Just getting back around to this. The missing config field has been corrected, along with json types and any new merge conflicts. Ive verified the validation test passes.

**Comment by SyntaxNode** [comment]
Cool. I've re-opened the PR. Looks like you need to push the new changes.

File: config/config.go [hunk 5]
```diff
@@ -518,6 +518,7 @@ func (cfg *Configuration) setDerivedDefaults() {
 	// openrtb_ext.BidderRTBHouse doesn't have a good default.
 	// openrtb_ext.BidderRubicon doesn't have a good default.
 	setDefaultUsersync(cfg.Adapters, openrtb_ext.BidderSharethrough, "https://match.sharethrough.com/FGMrCMMc/v1?redirectUri="+url.QueryEscape(externalURL)+"%2Fsetuid%3Fbidder%3Dsharethrough%26gdpr%3D{{.GDPR}}%26gdpr_consent%3D{{.GDPRConsent}}%26uid%3D%24UID")
+	setDefaultUsersync(cfg.Adapters, openrtb_ext.BidderSmartRTB, "https://market-global.smrtb.com/sync/all?nid=smartrtb&gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&rr="+url.QueryEscape(externalURL)+"%252Fsetuid%253Fbidder%253Dsmartrtb%2526gdpr%253D{{.GDPR}}%2526gdpr_consent%253D{{.GDPRConsent}}%2526uid%253D%257BXID%257D")
```
Comment (AndriyPavlyuk) [inline config/config.go]: Is it correct to decode twice ?

Comment (SyntaxNode) [reply config/config.go]: Totally depends on what happens on the SmartRTB side. @evanmsmrtb?

Comment (evanmsmrtb) [reply config/config.go]: This is expected, we double encode our return urls. Ive tested the sync url to ensure it is functional.

**Pull request merged**